src/main/java/org/reaktivity/k3po/nukleus/ext/internal/behavior/NukleusStreamFactory.java

```diff
@@ -163,10 +163,13 @@ private void onData(
                     channel.readExtBuffer(DATA).writeBytes(dataExtCopy);
                 }
 
-                if (channel.getConfig().getUpdate())
+                int padding = channel.getConfig().getPadding();
+                int window = channel.getConfig().getWindow();
+                partition.doWindow(channel, readableBytes + padding, padding);
+
+                if (!channel.getConfig().getUpdate() && readableBytes >= window)
                 {
-                    int padding = channel.getConfig().getPadding();
-                    partition.doWindow(channel, readableBytes + padding, padding);
+                    partition.doReset(streamId);
```

Comment:
These changes seem to alter the intended semantics of the code even though they might cause the specific test in question to now pass.

`update:none` means don't automatically send `WINDOW` when `DATA` is received.

Original code was correctly only sending `WINDOW` when `DATA` received if `update` is **not** `none`.

`channel.readableBytes()` should return the current number of inbound bytes allowed by flow control, and if it is exceeded then a `RESET` is sent by line 175 below.

When `DATA` frame arrives, `channel.readableBytes(...)` is decreased accordingly by line 151 above.

If flow control is active (`update` != `none`) then a `WINDOW` frame is sent to credit back the size of `DATA` received. The call to `partition.doWindow(...)` on line 169 increases `channel.readableBytes(...)` accordingly, but this does not occur when `update:none`.

The `channel.getConfig().getWindow()` value is used as the initial `WINDOW` credit sent to the remote stream endpoint, and represents the initial value of `channel.readableBytes()`.

The expectation is that as each `DATA` frame is sent by the remote stream endpoint, the `channel.readableBytes()` reduces accordingly, and is not re-credited due to `update:none` so eventually exceeds the initial `channel.getConfig().getWindow()` and triggers a `RESET`.

Please investigate the value of `channel.readableBytes()` when the `RESET` is expected but does not occur. May be related to `padding` being ignored when reducing `channel.readableBytes()`.